src/Chainweb/Pact/Types.hs

```diff
@@ -152,11 +153,20 @@ data PactServiceEnv = PactServiceEnv
   , _psPublicData :: !PublicData
   }
 
+-- data PactServiceState = PactServiceState
+--   {_psStateValidated :: Maybe BlockHeader
+--   }
+
 data PactServiceState = PactServiceState
-  { _psStateDb :: !PactDbState
-  , _psStateValidated :: !(Maybe BlockHeader)
+  { _psPactDbEnv :: PactDbEnv'
+  , _psStateValidated :: Maybe BlockHeader
```

Comment:
Let's move `_psPactDbEnv` out of state, it's dangerous as we could accidentally use the wrong instance. Instead, use the `PactDbEnv'` returned from `restore`, threaded through the various functions in `PactService`, and don't return this from the various initializer calls.

Reply:
done